### DISCUSSION TITLE
Time to compute cholesky (Eigen::LLT<MatrixXvar>)

### DISCUSSION THREAD

**Discussion by f-izzat** [autodiff/autodiff#221] [Q&A] [answered]
Im trying to invert an NxN matrix through the use of a cholesky decomposition. Essentially;

````
MatrixXvar X = ....... ;
Eigen::LLT<MatrixXvar> chol = X.llt();
````

However when trying to debug this takes quite a while? Is there a way to work around this?

**Comment by allanleal** [comment, marked as answer]
There is no workaround. Both Eigen and autodiff are C++ template libraries. In debug, the template codes in these libs are not inlined and optimized, and this results in significant overhead.
